src/main/groovy/edu/oregonstate/mist/advisors/resources/AdvisorsResource.groovy

```diff
@@ -0,0 +1,51 @@
+package edu.oregonstate.mist.advisors.resources
+
+import edu.oregonstate.mist.advisors.db.AdvisorDAO
+import edu.oregonstate.mist.api.AuthenticatedUser
+import edu.oregonstate.mist.api.Resource
+import edu.oregonstate.mist.api.jsonapi.ResourceObject
+import edu.oregonstate.mist.api.jsonapi.ResultObject
+import io.dropwizard.auth.Auth
+
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+import javax.ws.rs.Produces
+import javax.ws.rs.QueryParam
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+
+@Path('/advisors/')
+@Produces(MediaType.APPLICATION_JSON)
+class AdvisorsResource extends Resource {
+    private AdvisorDAO advisorDAO
+
+    AdvisorsResource(AdvisorDAO advisorDAO) {
+        this.advisorDAO = advisorDAO
+    }
+
+    @GET
+    Response getAdvisors(@Auth AuthenticatedUser _, @QueryParam('osuid') String osuid,
+                         @QueryParam('onid') String onid) {
+
+        if (onid && osuid) {
+            return badRequest("Please specify either the osuid or the onid, but not both.").build()
+        }
+        if (!onid && !osuid) {
+            return badRequest("Please specify either the osuid or the onid.").build()
+        }
+
+        def id = onid ?: osuid
+        def attributes = advisorDAO.getAdvisors(id)
+        def resultObject = new ResultObject()
+        attributes.each {
+            resultObject.data += new ResourceObject(
+                    id:         1, //@todo: what do we use for an id?
```

Comment:
I agree. I just had it hard coded since I didn't have a good solution for it. We don't want to expose the osu id of the supervisor since that would require extra permissions / MOU. Also the advisor id is not unique since multiple students may have the same advisor. I'm leaning towards using a hashCode of the object as the id.